cryptarchia/cryptarchia.py

```diff
@@ -167,8 +183,8 @@ def copy(self):
             block=self.block,
             nonce=self.nonce,
             total_stake=self.total_stake,
-            commitments=self.commitments.copy(),
-            nullifiers=self.nullifiers.copy(),
+            commitments=deepcopy(self.commitments),
+            nullifiers=deepcopy(self.nullifiers),
```

Comment:
these commitments/nullifiers are never changed no? I didn't think there was a risk to a shallow copy.

But better to be paranoid 👍

Reply:
We add the nullifier of the new leader proof, so yes that changes